webapp/platform/design-system/src/components/primitives/spinner/spinner.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+import type {HTMLAttributes} from 'react';
+
+import './spinner.scss';
+
+// Figma design system spinner sizes
+export type SpinnerSize = 10 | 12 | 16 | 20 | 24 | 28 | 32;
+
+export interface SpinnerProps extends Omit<HTMLAttributes<HTMLSpanElement>, 'role'> {
+
+    /** Size of the spinner in pixels (10, 12, 16, 20, 24, 28, 32) */
+    size?: SpinnerSize;
+
+    /** Whether the spinner should use inverted colors for dark backgrounds */
+    inverted?: boolean;
+
+    /** Accessible label for screen readers */
+    'aria-label'?: string;
+}
+
+const Spinner: React.FC<SpinnerProps> = ({
+    size = 16,
+    inverted = false,
+    className,
+    'aria-label': ariaLabel = 'Loading',
+    style,
+    ...htmlProps
+}) => {
+    // Calculate stroke width based on size (roughly 10% of size, with min/max bounds)
+    const strokeWidth = Math.max(1, Math.min(3, size * 0.1));
+
+    const spinnerClasses = classNames(
+        'Spinner',
+        {
+            'Spinner--inverted': inverted,
+        },
+        className,
+    );
+
+    const spinnerStyle = {
+        width: `${size}px`,
+        height: `${size}px`,
+        '--spinner-stroke-width': `${strokeWidth}px`,
+        ...style,
+    };
+
+    return (
+        <span
+            className={spinnerClasses}
+            role='status'
+            aria-label={ariaLabel}
+            aria-hidden='true'
```

Comment:
Note that these actually conflict. Setting `aria-hidden` causes the browser to ignore this element for accessibility purposes. We should remove that so that "Loading" is accessible.

Also, if you remove that, the `{hidden: true}`s that were added to the tests in this PR can be reverted

Reply:
Updated this and related tests